File: js-archive/check-if-one-string-swap-can-make-strings-equal.js
const areAlmostEqual = function (s1, s2) {
  let r = 0
  let m1 = ''
  let m2 = ''
  if (s1.length !== s2.length) {return false}


  for (let i = 0; i < s1.length; i++) {
    if (s1[i]!==s2[i]) {
      r++

      if (r === 1) {
        m1+= s1[i] + s2[i]
      }

      if (r === 2) {
        m2+= s2[i] + s1[i]
      }
    }
  }

  if (r === 0) {
    return true
  }

  if (r === 2) {
    return m1 === m2
  }

  return false
}

File: check-if-one-string-swap-can-make-strings-equal.go
func areAlmostEqual(s1 string, s2 string) bool {
    if (len(s1) != len(s2)) {return false}

    dif:=0;k1:="";k2:=""
    for i:=0; i < len(s1); i++ {
        if s1[i] != s2[i] {
            dif++
            k1+=string(s1[i])
            k2 = string(s2[i]) + k2
        }
    }

    if (dif == 0 || (dif == 2 && k1 == k2)) {return true}
    return false
}
